api/age-group/models/age-group.js

```diff
@@ -8,6 +8,59 @@ const { createLifecycleHooks } = require("../../../utils/algolia");
 
 const contentType = "age-group";
 
+const hooks = createLifecycleHooks(contentType);
+
+const updateActivityGroupActivities = async (group, ageGroupId) => {
+  const activityGroup = await strapi
+    .query("activity-group")
+    .findOne({ id: group.id });
+
+  if (activityGroup.activity_groups?.length) {
+    console.log(
+      "Activity group",
+      activityGroup.title,
+      "has",
+      activityGroup.activity_groups?.length,
+      "activity groups"
+    );
+    for (const g of activityGroup.activity_groups) {
+      await updateActivityGroupActivities(g, ageGroupId);
+    }
+  }
+
+  if (activityGroup.activities?.length) {
+    console.log(
+      "Activity group",
+      activityGroup.title,
+      "has",
+      activityGroup.activities?.length,
+      "activities"
+    );
+    for (const activity of activityGroup.activities) {
+      console.log("Updating age-group", ageGroupId, "to", activity.title);
+      await strapi.query("activity").update(
+        { id: activity.id },
+        {
+          age_group: ageGroupId,
+        }
+      );
+    }
+  }
+};
+
 module.exports = {
-  lifecycles: createLifecycleHooks(contentType),
+  lifecycles: {
+    async afterUpdate(result) {
+      if (result.activity_groups?.length) {
+        (async () => {
+          for (const group of result.activity_groups) {
+            await updateActivityGroupActivities(group, result.id);
+          }
+        })();
```

Comment:
Is the function wrapping necessary here?

Reply:
Yes. The `afterUpdate` is run synchronously when entry is updated, and so it blocks sending the response.
This way we can run the updates for the activity groups one at a time and not block the returning of the `afterUpdate`. 

The alternative would be to just not await the `updateActivityGroupActivities` calls, but then all of those would be started at the same time, and I think that might be a bit too much work for one moment.